src/lsdb/catalog/dataset/healpix_dataset.py

```diff
@@ -136,7 +136,9 @@ def _perform_search(
         """Performs a search on the catalog from a list of pixels to search in
 
         Args:
-            metadata (hc.catalog.Catalog | hc.catalog.MarginCatalog): The metadata of the hipscat catalog.
+            metadata (hc.catalog.Catalog | hc.catalog.MarginCatalog): The metadata of
```

Comment:
Could this be `HealpixCatalog` instead of `Catalog | MarginCatalog`?

Reply:
Unfortunately not, the HealpixDataset's `BaseCatalogInfo` does not have the ra/dec information required for filtering, only these subclasses do; mypy complains.